v1/plugins/logs/plugin.go

```diff
@@ -676,8 +676,8 @@ func (p *Plugin) flushDecisions(ctx context.Context) {
 	case <-done:
 		p.logger.Info("All decisions in buffer uploaded.")
 	case <-ctx.Done():
-		switch ctx.Err() {
-		case context.DeadlineExceeded, context.Canceled:
+		switch err := ctx.Err(); {
+		case errors.Is(err, context.DeadlineExceeded), errors.Is(err, context.Canceled):
```

Comment:
Catching wrapped errors? Would that happen for `context.Err()`? 🤔

Reply:
good point! I don't think so actually... I'll revert this